### PR TITLE
NullableCompareToNullRector: Don't trust phpdocs

### PR DESCRIPTION
**Pull request by staabm**

### CODE REVIEW

File: rules-tests/CodingStyle/Rector/If_/NullableCompareToNullRector/Fixture/fixture.php.inc [hunk 1]
```diff
@@ -1,23 +1,27 @@
 <?php
 
-/** @var stdClass|null $value */
```
Comment (staabm) [inline rules-tests/CodingStyle/Rector/If_/NullableCompareToNullRector/Fixture/fixture.php.inc:3]: adjusted the previous fixture, which demonstrated a wrong phpdoc based example.

File: rules/CodingStyle/Rector/If_/NullableCompareToNullRector.php [hunk 2]
```diff
@@ -84,43 +85,30 @@ public function refactor(Node $node): ?Node
...
-
-        // is number?
-        if ($staticType->isSuperTypeOf(new IntegerType())->yes()) {
-            return false;
-        }
+        $nativeType = TypeCombinator::removeNull($nativeType);
 
-        // is bool?
-        if ($staticType->isSuperTypeOf(new BooleanType())->yes()) {
+        if ($nativeType->isScalar()->yes()) {
```
Comment (staabm) [inline rules/CodingStyle/Rector/If_/NullableCompareToNullRector.php:108]: simplified all the separate checks to isScalar()

File: rules/CodingStyle/Rector/If_/NullableCompareToNullRector.php [hunk 3]
```diff
@@ -84,43 +85,30 @@ public function refactor(Node $node): ?Node
 
     private function isNullableNonScalarType(Expr $expr): bool
     {
-        $staticType = $this->getType($expr);
-        if ($staticType instanceof MixedType) {
-            return false;
-        }
+        $nativeType = $this->nodeTypeResolver->getNativeType($expr);
```
Comment (staabm) [inline rules/CodingStyle/Rector/If_/NullableCompareToNullRector.php:88]: the actual change to not trust phpdoc

File: rules-tests/CodingStyle/Rector/If_/NullableCompareToNullRector/Fixture/skip_phpdoc.php.inc [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+/** @var stdClass|null $value */
```
Comment (staabm) [inline rules-tests/CodingStyle/Rector/If_/NullableCompareToNullRector/Fixture/skip_phpdoc.php.inc:3]: 1:1 turned the previous "fixture.php.inc" into this skip-test

**Review by samsonasik — APPROVED** [review]

**Comment by samsonasik** [comment]
Let's give it a try, thank you @staabm

**Pull request merged**